packages/arbor-store/src/utilities.ts

```diff
@@ -95,7 +95,15 @@ export function isDetached<T extends object>(node: T): boolean {
   // but an ancestor has been detached.'
   if (!isNode(node)) return true
 
-  return !node.$tree.getLinkFor(node) && !node.$tree.getNodeFor<Node>(node)
+  const path = node.$tree.getPathFor(node)
+
+  if (!path) {
+    return true
+  }
+
+  return path.seeds.some(
+    (seed) => !node.$tree.getLinkFor(seed) && !node.$tree.getNodeFor<Node>(seed)
```

Comment:
Ensures a node is detached if:

1. itself is detached;
2. or any of its ancestors are detached (since the state is a Tree).